### DISCUSSION TITLE
Unable to generate CTE in version 1.4.0b3

### DISCUSSION THREAD

**Discussion by theonlypoi** [sqlalchemy/sqlalchemy#5988] [Usage Questions] [answered]
Hi,

I was trying to generate a CTE  (DB: Postgresql). However, it did not generate any SQL query. So I tried to reproduce that using a dummy table. In this case also, it failed to generate any query. 

```python
class Demo(Base):
    __tablename__ = "demo"

    demo_id = Column(Integer, primary_key=True)
    demo_name = Column(String(30), nullable=False)
    demo_info = Column(String(60))
    
demo = Demo.__table__
Base.metadata.create_all(tables=[demo])

cte = select(Demo.demo_id, Demo.demo_name).where(Demo.demo_id == 1).cte()
print(cte)  # Not generating any sql query
```
I am relatively new to SQLAlchemy. So not sure if I am missing something. 

Thanks!

**Comment by zzzeek** [comment, marked as answer]
this is a little bit of an API quirk, and it would be better if print(cte) displayed something so we should fix that.   but, the CTE itself is only part of an enclosing SELECT statement in practice so you'll see it if you select() from it:

    print(cte.select())
